Improve campaign guidance UX and replace numerical scores with descri…

ofisk opened the pull request:
…ptive states

- Merge Guidance Agent into Onboarding Agent to reduce redundancy
- Replace numerical readiness scores (e.g., 'Overall Score: 30') with encouraging descriptive states ('Taking Root', 'Growing Strong', 'Legendary')
- Add 9 campaign states with RPG-themed names for better user understanding
- Fix help button authentication and routing to provide personalized guidance
- Enhance welcome message with detailed LoreSmith capabilities and purple-themed styling
- Fix scrolling issues in welcome message component
- Remove markdown bolding from all UI text for proper JSX rendering
- Maintain internal scoring logic while hiding numerical values from users

ofisk committed:
Improve campaign guidance UX and replace numerical scores with descriptive states

- Merge Guidance Agent into Onboarding Agent to reduce redundancy
- Replace numerical readiness scores (e.g., 'Overall Score: 30') with encouraging descriptive states ('Taking Root', 'Growing Strong', 'Legendary')
- Add 9 campaign states with RPG-themed names for better user understanding
- Fix help button authentication and routing to provide personalized guidance
- Enhance welcome message with detailed LoreSmith capabilities and purple-themed styling
- Fix scrolling issues in welcome message component
- Remove markdown bolding from all UI text for proper JSX rendering
- Maintain internal scoring logic while hiding numerical values from users

aniham commented on src/components/chat/WelcomeMessage.tsx:
```diff
@@ -14,15 +14,63 @@ export function WelcomeMessage({
+              <div>
+                <div className="font-semibold text-base mb-2 flex items-center gap-2">
+                  <span className="bg-purple-100 dark:bg-purple-900/30 text-purple-800 dark:text-purple-200 px-2 py-1 rounded-md text-sm">
+                    🎯 Build Your Campaign Library
```
i'm not a fan of the emojis including the 👋🏼 welcome one. idk, just. they used to be cool until LLMs ruined them. emojis with llms are like f1 with netflix

ofisk replied:
Nah, I'm with you. will remove

aniham commented on src/components/help/HelpButton.tsx:
```diff
@@ -33,14 +33,24 @@ interface HelpResponse {
 
 interface HelpButtonProps {
   onActionClick: (action: string) => void;
+  onGuidanceRequest?: () => void;
```
is this right? it takes no params, returns void, and is optional lol

ofisk replied:
yeah, it triggers a chat message on behalf of the user which then causes the appropriate agent to respond, hence the void. and I think optional because it has a feedback but I'll remove that and make the prop required

aniham commented on src/routes/onboarding.ts:
```diff
@@ -1,30 +1,134 @@
+      guidanceMessage +=
+        "Upload PDFs, images, or documents to expand your inspiration library.\n\n";
+    } else {
+      guidanceMessage += `📊 Campaign Status\n`;
```
same comment on emojis here

aniham commented on src/routes/onboarding.ts:
```diff
@@ -1,30 +1,134 @@
+          action: "upload_resource",
+          title: "Upload Campaign Resources",
+          description:
+            "Add PDFs, images, or documents to expand your campaign library",
```
really want to move away from saying PDF anywhere that's not the file processor that checks for file types :)

ofisk replied:
should we specify files at all? ("images", I mean)

aniham replied:
yes, i thin you should specify the content rather than the type. character images, maps, playbooks, campaign props, ...

aniham commented on src/tools/campaign-context/assessment-tools.ts:
```diff
@@ -263,15 +267,41 @@ function generateReadinessSummary(assessment: {
+    return `Your campaign is ${campaignState}. Focus on the priority areas to level up your readiness.`;
+  } else {
+    return `Your campaign is ${campaignState}. Every great adventure starts with a single step!`;
+  }
```
we have two conditionals returning text based on scores and campaign state. i'd combine them -- easier to catch and maintain later if we want to change anything about them

ofisk replied:
did that in the other tech debt pr

aniham reviewed (approved):
left some style comments, looks great tho and can be iterated on later, including by me. good to merge

ofisk committed:
Remove emojis from help and guidance user-facing messages

- Remove emojis from WelcomeMessage.tsx component headers
- Remove emojis from onboarding.ts guidance messages
- Remove emojis from helpContent.ts utility functions
- Remove emojis from HelpSystem.tsx component messages
- Maintain clean, professional messaging throughout help flows

ofisk committed:
Simplify HelpButton by removing legacy fallback popup behavior

- Remove optional flag from onGuidanceRequest prop (now required)
- Remove legacy popup/modal UI code (100+ lines)
- Remove unused state (isLoading, helpData, showHelp)
- Remove unused imports (useState, JWT_STORAGE_KEY, API_CONFIG, Card, MemoizedMarkdown)
- Remove unused HelpResponse interface
- Component now simply triggers guidance request in chat - much cleaner

The HelpButton is now a simple trigger that delegates to the chat-based guidance system
